File: ch8/forkdemo1.c
#include<stdio.h>
#include<stdlib.h>
#include<unistd.h>

int main(int argc, char const *argv[])
{
    int ret_from_fork,mypid;

    mypid = getpid();
    printf("Before: my pid is %d\n",mypid);
    
    ret_from_fork = fork();

    sleep(1);
    printf("After: my pid is %d,fork() said is %d\n",getpid(),ret_from_fork);
    return 0;
}

File: ch8/ch8_src/src/bin/forkdemo1.rs
use std::{error::Error, thread::sleep, time::Duration};

use nix::unistd::{fork, getpid};

fn main()->Result<(),Box<dyn Error>>{
    let mypid = getpid();
    println!("Before: my pid is {}",mypid);

    let ret_from_fork = unsafe { fork() }?;

    sleep(Duration::from_secs(1));
    println!("After: my pid is {},fork() said is {:?}",getpid(),ret_from_fork); 
    Ok(())
}